### ISSUE TITLE
NC 14.0.3 directory listing broken

### ISSUE THREAD

**Issue by kubco2** [nextcloud/server#11837]
### Steps to reproduce
1. use some subdirectory to not lock yourself out
2. in subdir create new text file "%.txt"

### Expected behaviour
open text file
list files in directory

### Actual behaviour
only circle spinning
going out and in subdir do not help

I tried to find it in db filecache, it is there with name files/subdir/%.txt
It is also saved in data as %.txt

### Server configuration

**Operating system**:
centos7
**Web server:**
apache 2.4.6
**Database:**
mariadb 10.0.13
**PHP version:**
php 7.1
**Nextcloud version:** (see Nextcloud admin page)
14.0.3
**Updated from an older Nextcloud/ownCloud or fresh install:**
migrate path OC-9.1.8 > OC-10.0.8 > NC 12.0.12 > NC 13.0.7 NC 14.0.3

**Where did you install Nextcloud from:**
official site

**List of activated apps:**
no special apps, default settings
**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Client configuration
**Browser:**
firefox 62.0.2
**Operating system:**
Fedora 27

### Log
no errors in apache , nextcloud, javascript

**Comment by kubco2** [comment]
When I list directory it returns:
```
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"><d:response><d:href>/remote.php/dav/files/jakub/praca-doc/test_zmazat/test2/</d:href><d:propstat><d:prop><d:getlastmodified>Mon, 15 Oct 2018 12:15:59 GMT</d:getlastmodified><d:getetag>&quot;5bc484ff7c521&quot;</d:getetag><d:resourcetype><d:collection/></d:resourcetype><oc:fileid>1096307</oc:fileid><oc:permissions>RGDNVCK</oc:permissions><oc:size>2</oc:size><nc:has-preview>false</nc:has-preview><nc:mount-type></nc:mount-type><nc:is-encrypted>0</nc:is-encrypted><x1:share-permissions xmlns:x1="http://open-collaboration-services.org/ns">31</x1:share-permissions><oc:tags/><oc:favorite>0</oc:favorite><oc:comments-unread>0</oc:comments-unread><oc:owner-id>jakub</oc:owner-id><oc:owner-display-name>jakub</oc:owner-display-name><oc:share-types/></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><d:getcontenttype/><d:getcontentlength/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/files/jakub/praca-doc/test_zmazat/test2/%21.txt</d:href><d:propstat><d:prop><d:getlastmodified>Mon, 15 Oct 2018 12:11:07 GMT</d:getlastmodified><d:getetag>&quot;5fe763875650cf140b2ffc46cdfa52c8&quot;</d:getetag><d:getcontenttype>text/plain</d:getcontenttype><d:resourcetype/><oc:fileid>1096375</oc:fileid><oc:permissions>RGDNVW</oc:permissions><oc:size>1</oc:size><d:getcontentlength>1</d:getcontentlength><nc:has-preview>true</nc:has-preview><nc:mount-type></nc:mount-type><x1:share-permissions xmlns:x1="http://open-collaboration-services.org/ns">19</x1:share-permissions><oc:tags/><oc:favorite>0</oc:favorite><oc:comments-unread>0</oc:comments-unread><oc:owner-id>jakub</oc:owner-id><oc:owner-display-name>jakub</oc:owner-display-name><oc:share-types/></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><nc:is-encrypted/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/remote.php/dav/files/jakub/praca-doc/test_zmazat/test2/%25.txt</d:href><d:propstat><d:prop><d:getlastmodified>Mon, 15 Oct 2018 12:15:59 GMT</d:getlastmodified><d:getetag>&quot;e774ce5a1dee544a4798fca526625689&quot;</d:getetag><d:getcontenttype>text/plain</d:getcontenttype><d:resourcetype/><oc:fileid>1096377</oc:fileid><oc:permissions>RGDNVW</oc:permissions><oc:size>1</oc:size><d:getcontentlength>1</d:getcontentlength><nc:has-preview>true</nc:has-preview><nc:mount-type></nc:mount-type><x1:share-permissions xmlns:x1="http://open-collaboration-services.org/ns">19</x1:share-permissions><oc:tags/><oc:favorite>0</oc:favorite><oc:comments-unread>0</oc:comments-unread><oc:owner-id>jakub</oc:owner-id><oc:owner-display-name>jakub</oc:owner-display-name><oc:share-types/></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><nc:is-encrypted/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>

```

**Comment by kubco2** [comment]
Fixed #11744
